autosklearn/ensemble_building/builder.py

```diff
@@ -0,0 +1,925 @@
+        max_models_on_disc: int | float | None = 100,
+        performance_range_threshold: float = 0,
+        seed: int = 1,
+        precision: int = 32,
+        memory_limit: int | None = 1024,
+        read_at_most: int | None = None,
+        logger_port: int = logging.handlers.DEFAULT_TCP_LOGGING_PORT,
+        random_state: int | np.random.RandomState | None = None,
+    ):
+        """
+        Parameters
+        ----------
+        backend: Backend
+            backend to write and read files
+
+        dataset_name: str
+            name of dataset
+
+        task_type: int
+            type of ML task
+
+        metric: str
+            name of metric to compute the loss of the given predictions
+
+        ensemble_size: int = 10
+            maximal size of ensemble (passed to autosklearn.ensemble.ensemble_selection)
+
+        ensemble_nbest: int | float = 100
+
+            * int: consider only the n best prediction (> 0)
+
+            * float: consider only this fraction of the best, between (0, 1)
+
+            Both with respect to the validation predictions.
+            If performance_range_threshold > 0, might return less models
+
+        max_models_on_disc: int | float | None = 100
+           Defines the maximum number of models that are kept in the disc.
+           It defines an upper bound on the models that can be used in the ensemble.
+
+           * int: and dictates the max number of models to keep. (>= 1)
+
+           * float: it will be interpreted as the max megabytes allowed of disc space.
+           If the number of ensemble candidates require more disc space than
+           this float value, the worst models are deleted to keep within this budget.
+           Models and predictions of the worst-performing models will be deleted then.
+
+           * None: the feature is disabled.
+
+        performance_range_threshold: float = 0
+            Will at most return the minimum between ensemble_nbest models,
+            and max_models_on_disc. Might return less
+
+            Keep only models that are better than:
+
+                x = performance_range_threshold
+                x * dummy
+
+            E.g dummy=2, best=4, thresh=0.5 --> only consider models with loss > 3
+
+        seed: int = 1
+            random seed that is used as part of the filename
+
+        precision: int [16 | 32 | 64 | 128] = 32
+            precision of floats to read the predictions
+
+        memory_limit: int | None = 1024
+            memory limit in mb. If ``None``, no memory limit is enforced.
+
+        read_at_most: int | None = 5
+            read at most n new prediction files in each iteration
+
+        logger_port: int = DEFAULT_TCP_LOGGING_PORT
+            port that receives logging records
+
+        random_state: int | RandomState | None = None
+            An int or RandomState object used for generating the ensemble.
+        """
+        if isinstance(ensemble_nbest, int) and ensemble_nbest < 1:
```

Comment:
In python 3, the only reason to do so is for custom objects that support the `Integral` interface are used for `ensemble_nbest`, i.e. things that act like an `int`. As this is an internal thing, I wouldn't want to start using and worrying about custom number types.

In python 2, apparently you had to do so because of `long` and `int` being treated as to separate things.

I'm not really sure why Integral was used to be honest other than perhaps it's a relic from Python 2 days. This parameter doesn't seem any different from all the other places where checks for `int` are used?

Reply:
I just discussed this with @KEggensperger who added this in #809 and she said it's fine to use `int` here.